enos/modules/vault_agent/templates/set-up-approle-and-agent.sh

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env bash
+
+set -e
+
+binpath=${vault_install_dir}/vault
+
+fail() {
+  echo "$1" 1>&2
+  return 1
+}
+
+test -x "$binpath" || fail "unable to locate vault binary at $binpath"
+
+export VAULT_ADDR='http://127.0.0.1:8200'
+export VAULT_TOKEN='${vault_token}'
+
+$binpath auth disable approle || true
+
+approle_create_status=$($binpath auth enable approle)
+
+approle_status=$($binpath write auth/approle/role/agent-role secret_id_ttl=700h token_num_uses=1000 token_ttl=600h token_max_ttl=700h secret_id_num_uses=1000)
+
+ROLEID=$($binpath read --format=json auth/approle/role/agent-role/role-id   | jq -r '.data.role_id')
+
+if [[ "$ROLEID" == '' ]]; then
+  fail "expected ROLEID to be nonempty, but it is empty"
+fi
+
+SECRETID=$($binpath write -f --format=json  auth/approle/role/agent-role/secret-id  | jq -r '.data.secret_id')
+
+if [[ "$SECRETID" == '' ]]; then
+  fail "expected SECRETID to be nonempty, but it is empty"
+fi
+
+echo $ROLEID > /tmp/role-id
+echo $SECRETID > /tmp/secret-id
+
+cat > /tmp/vault-agent.hcl <<- EOM
```

Comment:
This is great stuff for future reference. Hopefully down the line we'll create an `enos_vault_agent` resource to handle all of this.

Reply:
Yeah, I think it probably makes sense to refactor this out if we need to re-use it (potentially with other configurables, like more or less retries, as a quick example). I mostly just adapted my local vault agent config and put it in-line.